File: 02-UML-Class-Diagrams/src/java/chapter02/Author.java
package chapter02;

/**
 * UML: Association target.
 * Book --> Author (Book knows its Author, Author exists independently)
 */
public class Author {
    private String name;
    private String email;

    public Author(String name, String email) {
        this.name = name;
        this.email = email;
    }

    public String getName() { return name; }
    public String getEmail() { return email; }

    @Override
    public String toString() {
        return name;
    }
}

File: 02-UML-Class-Diagrams/src/rust/author.rs
/// Association target: Book knows Author, Author exists independently
pub struct Author {
    name: String,
    email: String,
}

impl Author {
    pub fn new(name: &str, email: &str) -> Self {
        Author {
            name: name.to_string(),
            email: email.to_string(),
        }
    }

    pub fn name(&self) -> &str {
        &self.name
    }

    pub fn email(&self) -> &str {
        &self.email
    }
}
